### PR TITLE
Add Quest option "ignore-block-replace"

### PR DESCRIPTION
**Pull request by therbz**
Add quest option for block/place cheating "allow-break-place-cheating"
Also added party integration on the onBlockPlace side of things
Could do with some general code cleanup... (nested if's)

### CODE REVIEW

**Comment by therbz** [comment]
Just adding as a note: the reason I want to do this is to allow some quests to allow this such as breaking and placing (fully-grown) wheat, but for some quests to disallow this such as breaking ancient debris.

**Comment by PikaMug** [comment]
Thanks for the request! It looks like you still need to add a new selection to [OptionsPrompt.java](https://github.com/PikaMug/Quests/blob/main/core/src/main/java/me/blackvein/quests/convo/quests/options/OptionsPrompt.java) (OptionsGeneralPrompt class) for toggling in the Quest Editor. Also, not a huge fan of the word "cheating" since it could be done mistakenly or without malintent. How about "ignore-block-replace" default to `true`?

Once it's ready for review I'd appreciate if you could squash the commits but if you aren't familiar with how don't worry about it!

File: core/src/main/java/me/blackvein/quests/listeners/BlockListener.java [hunk 1]
```diff
@@ -93,64 +93,69 @@ public void onBlockBreak(final BlockBreakEvent evt) {
...
                         }
-                        if (currentStage.containsObjective(placeType)) {
-                            for (final ItemStack is : quester.getQuestData(quest).blocksPlaced) {
-                                if (evt.getBlock().getType().equals(is.getType()) && is.getAmount() > 0) {
-                                    ItemStack toPlace = new ItemStack(is.getType(), 64);
-                                    for (final ItemStack stack : currentStage.getBlocksToPlace()) {
-                                        if (ItemUtil.compareItems(is, stack, true) == 0) {
-                                            toPlace = stack;
+                        // Sorry for nested if
+                        if (!(quest.getOptions().canAllowBreakPlaceCheating())) {
```
Comment (PikaMug) [inline core/src/main/java/me/blackvein/quests/listeners/BlockListener.java:97]: Not worried about the nesting but please remove the comment. Also there's an unnecessary pair of parenthesis in this if statement.

**Comment by therbz** [comment]
> How about "ignore-block-replace" default to true?
Yep that's definitely a better name

File: core/src/main/java/me/blackvein/quests/listeners/BlockListener.java (hunk 1 above)
Comment (therbz) [reply core/src/main/java/me/blackvein/quests/listeners/BlockListener.java]: Done

**Comment by PikaMug** [comment]
> I wonder if this option should be false by default, considering that results in the same behaviour before this PR

We want the plugin to ignore blocks being replaced, so `true` seems right unless I'm mistaken? If you mean the behavior before #1953 the plugin _was_ supposed to ignore blocks replaced but the functionality was mistakenly broken/removed at some point (thanks again for correcting that).

If you're alright with clarifying optIgnoreBlockReplace to something like "Ignore blocks replaced once broken" this should be good to go!

**Comment by therbz** [comment]
Ah I see what has happened, I just interpreted "ignore-block-replace" wrong. I thought it meant to ignore when it happens, but it actually means to actively discard replaced blocks. Fixing now and will commit

**Comment by therbz** [comment]
I've removed the ! when checking canIgnoreBlockReplace in BlockListener, and updated the English lang files. I wonder if `canIgnoreBlocksReplaced` is a less confusing name?

**Pull request merged**